pkg/apis/pipeline/v1/result_validation.go

```diff
@@ -111,6 +111,53 @@ func (tr TaskResult) validateValue(ctx context.Context) (errs *apis.FieldError)
 	return errs
 }
 
+// Validate implements apis.Validatable
+func (sr StepResult) Validate(ctx context.Context) (errs *apis.FieldError) {
+	if !resultNameFormatRegex.MatchString(sr.Name) {
+		return apis.ErrInvalidKeyName(sr.Name, "name", fmt.Sprintf("Name must consist of alphanumeric characters, '-', '_', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my-name',  or 'my_name', regex used for validation is '%s')", ResultNameFormat))
+	}
+
+	switch {
+	case sr.Type == ResultsTypeObject:
+		errs := validateObjectStepResult(sr)
+		return errs
+	case sr.Type == ResultsTypeArray:
+		return nil
+	// Resources created before the result. Type was introduced may not have Type set
+	// and should be considered valid
```

Comment:
This comment doesn't quite make sense to me. Wouldn't the comment below make more sense? (e.g. if empty, treat as string?) Otherwise, reword.

Reply:
yes. I copy-pasted this mainly. I dropped that line.